### ISSUE TITLE
HTMLSelectElement options range incorrectly evaluated

### ISSUE THREAD

**Issue by alien9** [capricorn86/happy-dom#571]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `happy-dom@6.0.4` for the project I'm working on.

The <select> element raises an error to vitest because the options range is incorrectly evaluated by the selectedIndex getter.
I am aware that the problem was already fixed in main, but the 6.0.4 version installed by npm still displays the problem. 
Here is the diff that solved my problem:

```
diff
diff --git a/node_modules/happy-dom/lib/nodes/html-select-element/HTMLSelectElement.js b/node_modules/happy-dom/lib/nodes/html-select-element/HTMLSelectElement.js
index d14c844..28854a3 100644
--- a/node_modules/happy-dom/lib/nodes/html-select-element/HTMLSelectElement.js
+++ b/node_modules/happy-dom/lib/nodes/html-select-element/HTMLSelectElement.js
@@ -156,7 +156,7 @@ class HTMLSelectElement extends HTMLElement_1.default {
      * @param value Value.
      */
     set selectedIndex(value) {
-        if (this.options.length - 1 > value || value < 0) {
+        if (this.options.length - 1 < value || value < 0) {
             throw new DOMException_1.default('Select elements selected index must be valid', DOMExceptionNameEnum_1.default.indexSizeError);
         }
         this._options.selectedIndex = value;

```

**Comment by capricorn86** [comment]
Relates to #618

**Comment by IGx89** [comment]
This was fixed in v7.0.0 already through #549

**Comment by capricorn86** [comment]
I guess we can close this then.

There is also a PR on the way by @IGx89 that will improve this further.

https://github.com/capricorn86/happy-dom/pull/619
